lua_modules/ftp/ftpserver.lua

```diff
@@ -0,0 +1,163 @@
+-- a simple ftp server
+local file,net,pairs,print,string,table = file,net,pairs,print,string,table
+local ftp, ftp_data, ftp_srv
+do
+local createServer = function (user, pass)
+local data_fnc = nil
+ftp_data = net.createServer(net.TCP, 180)
+ftp_data:listen(20, function (s) if data_fnc then data_fnc(s) end end)
```

Comment:
Proposal to handle "early connect" scenario:
```lua
if data_fnc then data_fnc(s) else data_sock = s end
```

With a `local data_sock = nil` above.